### PR TITLE
[Chore] Add unit test

### LINKED ISSUE #8: [Chore] Add unit test
## Why

Need to add unit test
 
## Acceptance Criteria?

- Cover at least 70% of `viewmodel`
- Make sure all the test passed successfully

### PR DESCRIPTION
**Pull request by Shayokh144**
https://github.com/nimblehq/ci-cd-sample-ios/issues/8

## What happened

Need to add unit test for `WeatherViewModel`
 
## Insight

- Add unit test for case when city name is not provided
- Add unit test for case when valid city name is provided
- Add unit test for case when city name is invalid
- Add mock use case class and mock logic for checking invalid city name
- Remove unnecessary unit test file
 
## Proof Of Work

- All test passed
- `WeatherViewModel` coverage 100%

Screenshot 2022-10-21 at 10 05 31 AM

## Remark
- CI won't work as free credit of Bitrise is finished

### CODE REVIEW

**Review by vnntsu — APPROVED** [review]

File: WeatherTodayTests/Sources/Mocks/MockWeatherUseCase.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+//
+//  Created by Taher on 20/10/22.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+import RxCocoa
+
+
```
Comment (blyscuit) [inline WeatherTodayTests/Sources/Mocks/MockWeatherUseCase.swift]:
```suggestion
```

File: WeatherTodayTests/Sources/Specs/WeatherViewModelSpec.swift [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            context("when invalid city name is provided") {
+
+                beforeEach {
+                    viewModel.input.didFinishTyping(cityName: "A")
+                }
+
+                it("should show default text") {
+                    expect(temperatureText) == defaultTemperatureText
+                }
+            }
```
Comment (blyscuit) [inline WeatherTodayTests/Sources/Specs/WeatherViewModelSpec.swift]: I suggest to wrap this block inside another describe for better test structure.
```
describe("its didFinishTyping") {
// this block
}
```

File: WeatherTodayTests/Sources/Mocks/MockWeatherUseCase.swift (hunk 1 above)
Comment (Shayokh144) [reply WeatherTodayTests/Sources/Mocks/MockWeatherUseCase.swift]: fixed in this [commit](https://github.com/nimblehq/ci-cd-sample-ios/pull/22/commits/3d65e1179992cc9c7442660fdb88d9d668fb923c)

File: WeatherTodayTests/Sources/Specs/WeatherViewModelSpec.swift (hunk 2 above)
Comment (Shayokh144) [reply WeatherTodayTests/Sources/Specs/WeatherViewModelSpec.swift]: fixed in this [commit](https://github.com/nimblehq/ci-cd-sample-ios/pull/22/commits/3d65e1179992cc9c7442660fdb88d9d668fb923c)

**Review by blyscuit — APPROVED** [review]

**Pull request merged**